tests/transforms/test_classical_jacobian.py

```diff
@@ -0,0 +1,269 @@
+    for _x in x:
+        qml.RX(_x, wires=0)
+    for i in range(len(y)):
+        [qml.RY(_y, wires=1) for _y in y[i]]
+    return qml.expval(qml.PauliZ(0))
+
+
+perm_3 = ([2, 0, 1], [1, 2, 0, 3])
+
+
+def circuit_4(x, y):
+    for i in perm_3[0]:
+        qml.RX(x[i], wires=0)
+    for j in perm_3[1]:
+        qml.RY(y[j // 2, j % 2], wires=1)
+    return qml.expval(qml.PauliZ(0))
+
+
+def circuit_5(x, y, z):
+    for _x in x:
+        qml.RX(_x, wires=0)
+    qml.RZ(y[0, 1] * y[1, 0], wires=1)
+    qml.RY(z[0] + 0.2 * z[1] ** 2, wires=1)
+    return qml.expval(qml.PauliZ(0))
+
+
+circuits = [circuit_0, circuit_1, circuit_2, circuit_3, circuit_4, circuit_5]
+args = [(a,), (a, b), (x,), (x, y), (x, y), (x, y, z)]
+interfaces = ["jax", "autograd", "tf", "torch"]
+
+class_jacs = [
+    (np.ones(4),),
+    (
+        np.array(
+            [
+                np.cos(a),
+                1 / 3,
+                0.0,
+                0.0,
+            ]
+        ),
+        np.array([0.0, 0.0, 2 * b, -1 / (b ** 2)]),
+    ),
+    (np.eye(len(x)),),
+    (
+        np.vstack([np.eye(len(x)), np.zeros((4, 3))]),
+        np.vstack([np.zeros((3,) + y.shape), np.eye(np.prod(y.shape)).reshape(-1, *y.shape)]),
+    ),
+    (
+        np.vstack([np.eye(len(x)), np.zeros((4, 3))])[perm_3[0] + [3, 4, 5, 6]],
+        np.vstack(
+            [np.zeros((3,) + y.shape), np.eye(np.prod(y.shape))[perm_3[1]].reshape(-1, *y.shape)]
+        ),
+    ),
+    (
+        np.vstack([np.eye(len(x)), np.zeros((2, 3))]),
+        np.vstack(
+            [
+                np.zeros((3,) + y.shape),
+                np.array([[[0.0, y[1, 0]], [y[0, 1], 0.0]]]),
+                np.zeros((1,) + y.shape),
+            ]
+        ),
+        np.vstack([np.zeros((4, 4)), np.array([1, 0.4 * z[1], 0.0, 0.0])]),
+    ),
+]
+
+
+expected_outputs_without_argnums = {
+    "jax": [_jac[0] for _jac in class_jacs],
+    "autograd": class_jacs,
+    "tf": class_jacs,
+    "torch": class_jacs,
+}
+
+
+@pytest.mark.parametrize("i, circuit_args", enumerate(zip(circuits, args)))
+@pytest.mark.parametrize("interface", interfaces)
+def test_without_argnums(i, circuit_args, interface):
+    circuit, args = circuit_args
+    if interface == "tf":
+        tf = pytest.importorskip("tensorflow")
+        args = tuple((tf.constant(arg, dtype=tf.double) for arg in args))
+    elif interface == "torch":
+        torch = pytest.importorskip("torch")
+        args = tuple((torch.tensor(arg) for arg in args))
+    elif interface == "jax":
+        # Do not need the package but skip if JAX device not available
+        pytest.importorskip("jax")
+
+    dev = qml.device("default.qubit", wires=2)
+    qnode = qml.QNode(circuit, dev, interface=interface)
+    jac = classical_jacobian(qnode)(*args)
+    expected_jac = expected_outputs_without_argnums[interface][i]
+    if interface == "autograd" and all((np.isscalar(arg) for arg in args)):
+        expected_jac = qml.math.stack(expected_jac).T
```

Comment:
Wait, why is the autograd interface returning the transpose compared to the others?

Reply:
It is not, this is just because of how I hardcoded the expected Jacobians: They are tuples and for this special case of scalar variables and `interface==autograd`, `qml.math.stack` will produce and array with the selected Jacobians as _rows_. However, in the output of `classical_jacobian`, the _columns_ correspond to the parameters and the rows to the gates, so that the transpose is needed here :)